src/azure/client.rs

```diff
@@ -1018,7 +1018,7 @@ impl ListClient for Arc<AzureClient> {
         let mut response: ListResultInternal = quick_xml::de::from_reader(response.reader())
             .map_err(|source| Error::InvalidListResponse { source })?;
 
-        let token = response.next_marker.take();
+        let token = response.next_marker.take().filter(|x| !x.is_empty());
```

Comment:
I think this might have been a bug, which would potentially lead us to make a redundant extra list request, but this could also be an azurite specific quirk